tests/foreman/cli/test_provisioningtemplate.py

```diff
@@ -33,6 +35,79 @@ def module_os_with_minor():
     return entities.OperatingSystem(minor=randint(0, 10)).create()
 
 
+@pytest.mark.e2e
+@pytest.mark.no_containers
+def test_positive_end_to_end_crud(module_org, module_location, module_os, target_sat):
+    """Create a new provisioning template with several attributes, list, update them,
+       clone the provisioning template and then delete it
+
+    :id: 5f2e487c-07a5-423d-98ce-92ef1ad3b08d
+
+    :steps:
+        1. Create a provisioning template with several attributes.
+        2. Assert if all attributes are associated with the created template.
+        3. List the template.
+        4. Update the template and assert if it is properly updated.
+        5. Clone the template.
+        6. Assert if the cloned template contains all the attributes of parent template.
+        7. Delete the template.
+
+    :expectedresults: Template is created with all the given attributes, listed, updated,
+                      cloned and deleted.
+
+    :CaseImportance: High
+
+    :CaseAutomation: Automated
+    """
+    name = gen_string('alpha')
+    new_name = gen_string('alpha')
+    cloned_template_name = gen_string('alpha')
+    template_type = random.choice(constants.TEMPLATE_TYPES)
+    # create
+    template = make_template(
+        {
+            'name': name,
+            'audit-comment': gen_string('alpha'),
+            'description': gen_string('alpha'),
+            'locked': 'no',
+            'type': template_type,
+            'operatingsystem-ids': module_os.id,
+            'organization-ids': module_org.id,
+            'location-ids': module_location.id,
+        }
+    )
+
+    assert template['name'] == name
+    assert template['locked'] == 'no'
+    assert template['type'] == template_type
+    assert module_os.title in template['operating-systems']
+    assert module_org.name in template['organizations']
+    assert module_location.name in template['locations']
+    # list
+    template_list = target_sat.cli.Template.list({'search': f'name={name}'})
+    assert template_list[0]['name'] == name
```

Comment:
can we add an assert for RC after executing each command?

Reply:
The commands don't have RC after execution, but we have added expected assertions after command execution which should be enough.